### PR TITLE
Refine group index feature

### PR DESCRIPTION
**Pull request by scara82**
This PR:
- Add link in the header for visiting the "groups List" page;
- Add translations;
- Add a partial for the groups list's table  header.

Trello card: [https://trello.com/c/YfkrPxFj/46-must-have-as-an-admin-i-want-to-see-my-groups-list](url)

### CODE REVIEW

**Comment by apalski** [comment]
LGTM!

File: config/locales/en.yml [hunk 1]
```diff
@@ -40,18 +40,18 @@ en:
...
     edit:
       title: "Update family details"
     form:
       submit: "Submit"
   groups:
     new:
-      title: "Add a new group"
+      title: "Add a group"
       start_date_in_the_past_error: "from date can not be in the past!"
       start_date_after_end_date_error: "to date must be after from date!"
```
Comment (elle) [inline config/locales/en.yml:54]: These two errors are not specific to only new groups. A suggestion to make them a bit more general.

Looking at Rails documentation: http://guides.rubyonrails.org/i18n.html#translations-for-active-record-models
Section 4.5.1, a suggestion to move these messages to:
```rb
activerecord:
  errors:
    models:
      group:
        start_date_in_the_past: ...
        start_date_after_end_date: ...
```

File: app/controllers/families_controller.rb [hunk 2]
```diff
@@ -17,12 +17,6 @@ def create
     respond_with @family, location: -> { families_path }
   end
 
-  def destroy
-    family.delete
-
-    respond_with @family, location: -> { families_path }
-  end
-
```
Comment (elle) [inline app/controllers/families_controller.rb:25]: This does not belong in this PR. Did you rebase with master to get the latest changes?

File: app/models/group.rb [hunk 3]
```diff
@@ -12,22 +12,20 @@ class Group < ApplicationRecord
   private
 
   def start_date_after_end_date
-    if start_date.blank? || end_date.blank?
-      return
-    end
-
-    if end_date <= start_date
-      t(".group.new.start_date_after_end_date_error")
+    if start_date.present? && end_date.present? && (end_date <= start_date)
```
Comment (elle) [inline app/models/group.rb]: Could this line change to: `if start_date && end_date && (end_date <= start_date)`?

File: app/models/group.rb [hunk 4]
```diff
@@ -12,22 +12,20 @@ class Group < ApplicationRecord
...
   end
 
   def start_date_in_the_past
-    if start_date.blank? || end_date.blank?
-      return
-    end
-
-    if start_date < Date.today
-      t(".group.new.start_date_in_the_past_error")
+    if start_date.present? && start_date < Date.today
```
Comment (elle) [inline app/models/group.rb]: Same here. Is `.present?` needed?

File: spec/features/admin_views_all_groups_spec.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+require "rails_helper"
+
+feature "Admin can see all the groups in the groups/index page" do
+  scenario "successfully" do
+    visit groups_path
```
Comment (elle) [inline spec/features/admin_views_all_groups_spec.rb]: I would create a group in this spec, and check that I can see its details

File: app/models/group.rb (hunk 3 above)
Comment (elle) [reply app/models/group.rb]: Are there specs for this validation?

File: app/models/group.rb (hunk 4 above)
Comment (elle) [reply app/models/group.rb]: And same question: are there specs for this validation rule?

File: app/views/groups/_groups_table_header.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<tr>
+  <th><%= Group.human_attribute_name(:name)%></th>
```
Comment (elle) [inline app/views/groups/_groups_table_header.html.erb]: Space between `)%` on all the lines

File: spec/features/admin_adds_a_group_spec.rb [hunk 7]
```diff
@@ -9,7 +9,7 @@
 
     fill_in Group.human_attribute_name(:name), with: "St. Peter"
     fill_in Group.human_attribute_name(:responsable_name), with: "Mr. Black"
-    fill_in Group.human_attribute_name(:responsable_contact), with: "0045-123-4445"
+    fill_in Group.human_attribute_name(:responsable_contact), with: "0045-1234"
```
Comment (elle) [inline spec/features/admin_adds_a_group_spec.rb]: A question: why is this change here? is there a reason?

File: config/locales/en.yml (hunk 1 above)
Comment (scara82) [reply config/locales/en.yml:54]: Done in sha 83fd409

File: app/models/group.rb (hunk 3 above)
Comment (scara82) [reply app/models/group.rb]: Yes there is a spec for testing `start_date_in_the_past` in the model/group_spec.rb. I just added the spec for the `start_date_after_end_date`.

Done in sha a80758b

I also removed the `present?` method in sha 83fd409

File: spec/features/admin_adds_a_group_spec.rb (hunk 7 above)
Comment (scara82) [reply spec/features/admin_adds_a_group_spec.rb]: Yes, Rubocop was complaining because too long :)

File: app/views/groups/_groups_table_header.html.erb (hunk 6 above)
Comment (scara82) [reply app/views/groups/_groups_table_header.html.erb]: Done in `75129ff`

File: spec/features/admin_views_all_groups_spec.rb (hunk 5 above)
Comment (scara82) [reply spec/features/admin_views_all_groups_spec.rb]: Done in sha: `67fc3ea`

File: app/controllers/families_controller.rb (hunk 2 above)
Comment (scara82) [reply app/controllers/families_controller.rb:25]: I think I deleted it because I realised that didn't belong to this PR.. :sweat_smile:

File: spec/features/admin_adds_a_group_spec.rb (hunk 7 above)
Comment (elle) [reply spec/features/admin_adds_a_group_spec.rb]: I would use the field's id (i.e. `group_responsible_contact` instead of `Group.human_attribute_name(:responsable_contact)` and then use input that is close to reality.

Which brings the question: should you have a validation on phone number? for example, that it has to be between 9-10 characters?

**Comment by scara82** [comment]
@elle all changes are implemented, it would be great if you can have a look again. Thanks!

File: app/controllers/families_controller.rb (hunk 2 above)
Comment (elle) [reply app/controllers/families_controller.rb:25]: 👍

**Review by elle — APPROVED** [review]
Besides 1 comment, LGTM. Merge at will

**Comment by scara82** [comment]
@elle  and @apalski  thanks for your feedbacks. I'm about to merge it!

**Comment by scara82** [comment]
Thanks @elle  and @apalski for your feedbacks. I'm merging now!
